src/mixins/keyboard-shortcut-mixin.ts

```diff
@@ -1,26 +1,69 @@
 import { LitElement } from "lit-element";
 import { Constructor } from "../types";
 
+type RegisteredShortcuts = {
+  [key in HAKeyboardShortcut]?: () => void;
+};
+
+export enum HAKeyboardShortcut {
+  CTRL_S = "CtrlS",
+  CTRL_P = "CtrlP",
+  CTRL_SHIFT_P = "CtrlShiftP",
+}
+
+const isMac = /(Mac|iPhone|iPod|iPad)/i.test(navigator.platform);
+
 export const KeyboardShortcutMixin = <T extends Constructor<LitElement>>(
   superClass: T
 ) =>
   class extends superClass {
-    private _keydownEvent = (event: KeyboardEvent) => {
-      if ((event.ctrlKey || event.metaKey) && event.key === "s") {
-        event.preventDefault();
-        this.handleKeyboardSave();
-      }
-    };
-
-    public connectedCallback() {
-      super.connectedCallback();
-      this.addEventListener("keydown", this._keydownEvent);
-    }
+    private _registeredShortcuts: RegisteredShortcuts = {};
 
     public disconnectedCallback() {
       this.removeEventListener("keydown", this._keydownEvent);
```

Comment:
We remove the listener on disconnected, we should add it again on connected. The firstupdated will only run once, so after the element is disconnected the listener will never be added again on connect

Reply:
Also, this will not remove it now, as the callbacks for the add and remove are different.